src/object/obj_class.c

```diff
@@ -491,10 +677,14 @@ obj_ec_codec_init()
 
 	i = 0;
 	for (oc = &daos_obj_classes[0]; oc->oc_id != OC_UNKNOWN; oc++) {
-		if (oc->oc_attr.ca_resil != DAOS_RES_EC)
+		if (oc->oc_attr.ca_resil != DAOS_RES_EC ||
+		    oc->oc_attr.ca_grp_nr != 1 ||
```

Comment:
same, grp_nr != 1 check looks not very good, we may check if k&p corresponding code if inited already. We may define DAOS_EC_MAXK / DAOS_EC_MAXP (maybe 32 and 8 respectively?), and the oc_ec_codecs is an array with MAXK*MAXP items and each item (daos_oc_ec_codec) with one inited flag.
then the obj_ec_codec_get() is very simple that just calculate its index of the array, and fini is simple that just fini it if the inited flag is set.  Just FYI.